lib/rules/padding-line-between-tags.js

```diff
@@ -109,40 +157,23 @@ function checkNewline(context) {
         const lineDifference =
           closestSibling.loc.start.line - endTag.loc.end.line
         if (configure.blankLine === 'always') {
-          if (lineDifference === 1) {
-            insertNewLine(context, block, closestSibling)
-          } else if (lineDifference === 0) {
-            context.report({
-              messageId: 'always',
-              loc: closestSibling.loc,
-              // @ts-ignore
-              fix(fixer) {
-                const lastSpaces = /** @type {RegExpExecArray} */ (
-                  /^\s*/.exec(
-                    context.getSourceCode().lines[endTag.loc.start.line - 1]
-                  )
-                )[0]
-
-                return fixer.insertTextAfter(endTag, `\n\n${lastSpaces}`)
-              }
-            })
+          insertNewLine(context, block, closestSibling, lineDifference)
+        } else if (configure.blankLine === 'consistent') {
+          const siblingElements = block.parent.children.filter(
+            (element) =>
+              element.type === 'VElement' &&
+              hasConsistentConfiguration(element, configureList)
+          )
+
+          const siblingLineDifference =
+            siblingElements[1].loc.start.line - siblingElements[0].loc.end.line
+          if (siblingLineDifference > 1) {
+            insertNewLine(context, block, closestSibling, lineDifference)
+          } else {
+            removeExcessLines(context, endTag, closestSibling, lineDifference)
           }
```

Comment:
I think that branching can be simplified by rewriting the logic beyond L150 as follows.


```js
    const configure = reverseConfigureList.find(
      (configure) =>
        (configure.prev === '*' || block.name === configure.prev) &&
        (configure.next === '*' || closestSibling.name === configure.next)
    )

    if (!configure) {
      return
    }
    const lineDifference = closestSibling.loc.start.line - endTag.loc.end.line

    let blankLine = configure.blankLine
    if (blankLine === 'consistent') {
      const firstConsistentBlankLine = firstConsistentBlankLines.get(
        block.parent
      )
      if (firstConsistentBlankLine == null) {
        // If don't have a style for the first `blankLine="consistent"` yet, set the style and finish checking.
        firstConsistentBlankLines.set(
          block.parent,
          lineDifference > 1 ? 'always' : 'never'
        )
        return
      }
      blankLine = firstConsistentBlankLine
    }

    if (blankLine === 'always') {
      insertNewLine(context, block, closestSibling, lineDifference)
    } else {
      removeExcessLines(context, endTag, closestSibling, lineDifference)
    }
```

For this we need to add the following variable declarations to L127.

```js
  const reverseConfigureList = [...configureList].reverse()

  /**
   * It has the style of the first `blankLine="consistent"`.
   * @type {Map<VElement, "always" | "never">}
   */
  const firstConsistentBlankLines = new Map()
```